atlp-rwanda/falcons-ec-be
# #184637265 ft(): add product to wishlist

**What does this PR do?**
A buyer should be able to add a product to a wish list
**Description of Task to be completed?**

- The backend needs to handle the addition of items to the buyer's wishlist, including storing the wishlist in the buyer's account and sending a confirmation message to the frontend. \
- As a buyer I should be able to list all  wished products by me, as a seller i should be able to list all products from my collections that are wished

**How should this be manually tested?**

- Clone the repository
- Checkout ft-product-wishlist-184637265 branch
- Run npm install to install the dependencies and dev dependencies in your terminal
- Run npm run dev to start the server in the terminal and check if the server is working
- In the URL field enter localhost: PORT/API-docs to open Swagger UI
- Login users(buyer) using /api/v1/users/signin endpoint
- Add a product to product wishlist using /api/v1/productWish endpoint
- Retrieve all buyer product wishes /api/v1/productWishes 
- Login user(seller) using /api/v1/users/signin
- Retrieve all wished product in a seller's collection using  /api/v1/productWishes 
- Retrieve all product wishes per product using  /api/v1/{product_id}/productWishes  

**Any background context you want to provide?** 
none 
**What are the relevant pivotal tracker/Trello stories?**
 tracker: #184637265
Screenshots (if appropriate)
 none 
Questions: 
none
